src/bin/console/console.cpp

```diff
@@ -536,6 +543,42 @@ int Console::_generate_tpcds(const std::string& args) {
   return ReturnCode::Ok;
 }
 
+int Console::_generate_ssb(const std::string& args) {
+  const auto arguments = tokenize(args);
+
+  if (arguments.empty() || arguments.size() > 2) {
+    out("Usage: ");
+    out("  generate_ssb SCALE_FACTOR [CHUNK_SIZE]   Generate SSB tables with the specified scale factor. \n");
+    out("                                           Chunk size is " + std::to_string(Chunk::DEFAULT_SIZE) +
+        " by default. \n");
+    return ReturnCode::Error;
+  }
+
+  const auto scale_factor = std::stof(arguments.at(0));
```

Comment:
Why parse with with stof here but later with boost lexical cast?

Reply:
Should not be a difference here, according to [Stack Overflow](https://stackoverflow.com/questions/23582089/is-boostlexical-cast-redundant-with-c11-stoi-stof-and-family). `boost::lexical_cast` seems to be way more powerful, but that's nothing we need here.

Reply:
They also parse 1e17 etc. So leave it?

Reply:
I tries some things with `lexical_cast` and STL:
```
(release)> generate_ssb 0.01 1000
SCALE_FACTOR
  std::stof 0.01
  boost::lexical_cast 0.01
CHUNK_SIZE
  std::stoul 1000
  boost::lexical_cast 1000

(release)> generate_ssb 0,01 1,000
SCALE_FACTOR
  std::stof 0
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::bad_lexical_cast>: bad lexical cast: source type value could not be interpreted as target
CHUNK_SIZE
 std::stoul 1
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::bad_lexical_cast>: bad lexical cast: source type value could not be interpreted as target

(release)> generate_ssb 10'000 100'000
SCALE_FACTOR
 std::stof 10
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::bad_lexical_cast>: bad lexical cast: source type value could not be interpreted as target
CHUNK_SIZE
 std::stoul 100
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::bad_lexical_cast>: bad lexical cast: source type value could not be interpreted as target

(release)> generate_ssb 1e1 1e2
SCALE_FACTOR
  std::stof 10
  boost::lexical_cast 10
CHUNK_SIZE
  std::stoul 1
libc++abi: terminating with uncaught exception of type boost::wrapexcept<boost::bad_lexical_cast>: bad lexical cast: source type value could not be interpreted as target

```

So actually boost is more explicit by throwing errors while the STL yields garbage when in doubt, which is by definition (`std::toul`):
> Discards any whitespace characters (as identified by calling [std::isspace](https://en.cppreference.com/w/cpp/string/byte/isspace)) until the first non-whitespace character is found, then takes as many characters as possible to form a valid base-n (where n=base) unsigned integer number representation and converts them to an integer value.

Thus, I changed all casts to `boost::lexical` cast for all benchmarks.